.github/workflows/build-and-test-differential.yaml

```
@@ -57,6 +62,7 @@ jobs:
           fetch-depth: ${{ env.PR_FETCH_DEPTH }}
 
       - name: Run build-and-test-differential action
+        if: ${{ !(matrix.container-suffix == '-cuda') || needs.make-sure-require-cuda-label-is-present.outputs.result == 'true' }}
```

I think it is AND condition.

```suggestion
        if: ${{ !(matrix.container-suffix == '-cuda') && needs.make-sure-require-cuda-label-is-present.outputs.result == 'true' }}
```

---

I want to set the value to false only when the suffix is "cuda" and the label is not present. In all other cases, I want to set it to true. Below is the truth table I want to achieve:
| | label is present | label is not present |
| -------- | -------- | -------- |
| suffix == "cuda" | true | false |
| !(suffix == "cuda") | true | true |

I think we can achieve the desired logic using an OR operation. The expression `!(matrix.container-suffix == '-cuda')` ensures that the condition is true when the suffix is not `'cuda'`. The expression `needs.make-sure-require-cuda-label-is-present.outputs.result == 'true'` ensures that the condition is true when the label is present.

---

Oh, I miss the `!` mark. Now I understood.